airtel/transport.go

```diff
@@ -0,0 +1,122 @@
+//roundTripper this override the default RoundTripper of net/http package
+package airtel
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"time"
+)
+
+// Supported authentication schemes.
+const (
+	SchemeBearer = "Bearer"
+	SchemeToken  = "token"
+)
+
+//Transport  is an http.RoundTrip that refreshes oauth
+// tokens, wrapping a base Transport and refreshing the
+// token if expired with max retry.
+//request that require authorization header by appending header on it roundTripper
+type Transport struct {
```

Comment:
This is a retry transport right? Be specific about it then

Reply:
okay! sure i will explain on it in the next changes

Reply:
I meant the name should be something like reTry transport, also there some good retry backoff libraries out there you can use: https://github.com/superfly/flyctl/blob/0247e3ef96acc66949a397a0e74820633cf4d2a5/retry/retry.go#L9-L30